mmdet/models/dense_heads/yolo_head.py

```diff
@@ -0,0 +1,652 @@
+        self.balance_conf = balance_conf
+
+        # self.num_attrib = num_classes + bboxes (4) + objectness (1)
+        self.num_attrib = self.num_classes + 5
+        self.last_layer_dim = self.num_anchors_per_scale * self.num_attrib
+
+        cfg = dict(conv_cfg=conv_cfg, norm_cfg=norm_cfg, act_cfg=act_cfg)
+        self.convs_bridge = nn.ModuleList()
+        self.convs_final = nn.ModuleList()
+        for i_scale in range(self.num_scales):
+            in_c = self.in_channels[i_scale]
+            out_c = self.out_channels[i_scale]
+            conv_bridge = ConvModule(in_c, out_c, 3, padding=1, **cfg)
+            conv_final = nn.Conv2d(out_c, self.last_layer_dim, 1, bias=True)
+
+            self.convs_bridge.append(conv_bridge)
+            self.convs_final.append(conv_final)
+
+        self.train_cfg = train_cfg
+        self.test_cfg = test_cfg
+
+    def init_weights(self):
+        for m in self.modules():
+            if isinstance(m, nn.Conv2d):
+                xavier_init(m, distribution='uniform')
+
+    def forward(self, feats):
+        assert len(feats) == self.num_scales
+        results = []
+        for i in range(self.num_scales):
+            x = feats[i]
+            x = self.convs_bridge[i](x)
+            out = self.convs_final[i](x)
+            results.append(out)
+
+        return tuple(results),
+
+    @force_fp32(apply_to=('results_raw', ))
+    def get_bboxes(self, results_raw, img_metas, cfg=None, rescale=False):
+        """Transform network output for a batch into bbox predictions.
+
+        Args:
+            results_raw (list[Tensor]): Raw predictions for a batch of images.
+            img_metas (list[dict]): Meta information of each image, e.g.,
+                image size, scaling factor, etc.
+            cfg (mmcv.Config): Test / postprocessing configuration,
+                if None, test_cfg would be used
+            rescale (bool): If True, return boxes in original image space
+
+        Returns:
+            list[tuple[Tensor, Tensor]]: Each item in result_list is 2-tuple.
+                The first item is an (n, 5) tensor, where the first 4 columns
+                are bounding box positions (tl_x, tl_y, br_x, br_y) and the
+                5-th column is a score between 0 and 1. The second item is a
+                (n,) tensor where each item is the predicted class label of the
+                corresponding box.
+        """
+        result_list = []
+        for img_id in range(len(img_metas)):
+            result_raw_list = [
+                results_raw[i][img_id].detach() for i in range(self.num_scales)
+            ]
+            scale_factor = img_metas[img_id]['scale_factor']
+            proposals = self.get_bboxes_single(result_raw_list, scale_factor,
+                                               cfg, rescale)
+            result_list.append(proposals)
+        return result_list
+
+    @staticmethod
+    def _get_anchors_grid_xy(num_grid_h, num_grid_w, stride, device='cpu'):
```

Comment:
In dense heads, we usually use `anchor_generator` and `box_coder` to generate anchors and encode/decode boxes. YOLOHead should also do that and may implement a new anchor generator and box coder for its special needs. Thus the anchor and box coder could also be used for more general usage.

Reply:
Hi @ZwwWayne , 

I agree that implementing a new anchor generator and bbox (de)coder is reasonable, but it will introduce a refactor of the current YOLOHead, which might be at a cost too high for this single pr. Would you accept that we introduce the generator and coder as an incremental work in a new pr, after the main code is merged?